src/pages/home/report/ReportActionCompose.js

```diff
@@ -471,27 +495,73 @@ class ReportActionCompose extends React.Component {
 
         // the larger composerHeight the less space for EmojiPicker, Pixel 2 has pretty small screen and this value equal 5.3
         const hasEnoughSpaceForLargeSuggestion = this.props.windowHeight / this.state.composerHeight >= 6.8;
-        const isEmojiPickerLarge = !this.props.isSmallScreenWidth || (this.props.isSmallScreenWidth && hasEnoughSpaceForLargeSuggestion);
+        const isAutoSuggestionPickerLarge = !this.props.isSmallScreenWidth || (this.props.isSmallScreenWidth && hasEnoughSpaceForLargeSuggestion);
 
         const nextState = {
             suggestedEmojis: [],
             highlightedEmojiIndex: 0,
             colonIndex,
-            shouldShowSuggestionMenu: false,
-            isEmojiPickerLarge,
+            shouldShowEmojiSuggestionMenu: false,
+            isAutoSuggestionPickerLarge,
         };
         const newSuggestedEmojis = EmojiUtils.suggestEmojis(leftString);
 
         if (newSuggestedEmojis.length && isCurrentlyShowingEmojiSuggestion) {
             nextState.suggestedEmojis = newSuggestedEmojis;
-            nextState.shouldShowSuggestionMenu = !_.isEmpty(newSuggestedEmojis);
+            nextState.shouldShowEmojiSuggestionMenu = !_.isEmpty(newSuggestedEmojis);
         }
 
         LayoutAnimation.configureNext(LayoutAnimation.create(50, LayoutAnimation.Types.easeInEaseOut, LayoutAnimation.Properties.opacity));
 
         this.setState(nextState);
     }
 
+    calculateMentionSuggestion() {
+        if (this.state.selection.end < 1) {
```

Comment:
@szebniok This change caused [this bug](https://github.com/Expensify/App/issues/18879). Could you help to explain why we need to add this?  Could we use `!this.state.value` instead of `this.state.selection.end ` < 1?

Reply:
@dukenv0307 this is how it's done for the existing emoji suggestion feature, so I'm guessing we just replicated that behavior for mentions.

Reply:
We should have reset the `resetSuggestions` as we do for Emoji Suggestion but we missed that here